### PR TITLE
aws account schema changes

### PR DESCRIPTION
**Pull request by shireesh-illumio**
Changing plan modifiers to the fields to work with current illumio apis.

### CODE REVIEW

File: api/schema/aws_account_resource.go
Comment (rlenglet) [inline api/schema/aws_account_resource.go:1]: We decided to now support AWS organizations for now. Delete everything about the `aws_organization` resource:
 * `api/schema/aws_organization_resource.go`
 * `examples/resources/illumio-cloudsecure_aws_organization`
 * the resource in the list of resources in the schema, at https://github.com/illumio/terraform-provider-illumio-cloudsecure/blob/0bc6af692610b7e69fcf4dcfb8831e13dca9aa9b/api/schema/schema.go#L21

File: api/schema/aws_account_resource.go [hunk 1]
```diff
@@ -32,7 +32,7 @@ var (
 						Mode: ImmutableAttributeMode,
 					},
 				},
-				"mode": StringResourceAttributeWithMode{
+				"access_mode": StringResourceAttributeWithMode{
```
Comment (rlenglet) [inline api/schema/aws_account_resource.go:35]: 1. Update the example(s) whenever you change a schema. https://github.com/illumio/terraform-provider-illumio-cloudsecure/blob/0bc6af692610b7e69fcf4dcfb8831e13dca9aa9b/examples/resources/illumio-cloudsecure_aws_account/resource.tf#L9
1. Re-generate the doc afterwards.

File: api/schema/aws_account_resource.go [hunk 2]
```diff
@@ -52,6 +52,9 @@ var (
 				"name": resource_schema.StringAttribute{
 					Description: "Display name for the AWS account.",
 					Required:    true,
+					Validators: []validator.String{
+						stringvalidator.LengthAtLeast(3),
+					},
```
Comment (rlenglet) [inline api/schema/aws_account_resource.go:57]: Remove. Where does this come from? I can't find any such validation on the server side.
```suggestion
```

File: api/schema/aws_account_resource.go [hunk 3]
```diff
@@ -62,7 +65,7 @@ var (
 						},
 					},
 					attributeWithMode: attributeWithMode{
-						Mode: ImmutableAttributeMode,
+						Mode: WriteOnlyOnceAttributeMode,
```
Comment (rlenglet) [inline api/schema/aws_account_resource.go:68]: Revert those changes from `ImmutableAttributeMode` to `WriteOnlyOnceAttributeMode`.
`ImmutableAttributeMode` is the correct mode here.
`WriteOnlyOnceAttributeMode` is meant to be used for immutable secrets set by the client, which should never be returned by the server. This is not the case here.
```suggestion
						Mode: ImmutableAttributeMode,
```

File: api/schema/aws_account_resource.go [hunk 4]
```diff
@@ -74,7 +77,7 @@ var (
...
```
Comment (rlenglet) [inline api/schema/aws_account_resource.go:80]:
```suggestion
						Mode: ImmutableAttributeMode,
```

File: api/schema/aws_account_resource.go [hunk 5]
```diff
@@ -86,7 +89,7 @@ var (
...
```
Comment (rlenglet) [inline api/schema/aws_account_resource.go:92]:
```suggestion
						Mode: ImmutableAttributeMode,
```

File: api/schema/aws_account_resource.go (hunk 1 above)
Comment (rlenglet) [inline api/schema/aws_account_resource.go:35]: More generally, why is this change necessary?

File: api/schema/aws_account_resource.go [hunk 6]
```diff
@@ -52,6 +52,9 @@ var (
 				"name": resource_schema.StringAttribute{
 					Description: "Display name for the AWS account.",
 					Required:    true,
+					Validators: []validator.String{
+						stringvalidator.LengthAtLeast(3),
+					},
 				},
 				"organization_master_account_id": StringResourceAttributeWithMode{
```
Comment (rlenglet) [inline api/schema/aws_account_resource.go:59]: Remove this `organization_master_account_id` attribute. We have decided to remove support for organizations.
And when we get around to supporting that, we shouldn't refer to the organization by master account ID.

Comment (rlenglet) [inline api/schema/aws_account_resource.go:59]: Add an `organization_id` attribute. We won't use it for now on the server side, but when we get around to supporting grouping accounts by organization we'll already have the data for each account.
```
				"organization_id": StringResourceAttributeWithMode{
					StringAttribute: resource_schema.StringAttribute{
						Description: "ID of the AWS organization.",
						Optional:    true,
						PlanModifiers: []planmodifier.String{
							stringplanmodifier.RequiresReplace(),
						},
					},
					attributeWithMode: attributeWithMode{
						Mode: ImmutableAttributeMode,
					},
				},
```

**Review by rlenglet — DISMISSED** [review]

File: api/schema/aws_account_resource.go (hunk 2 above)
Comment (shireesh-illumio) [reply api/schema/aws_account_resource.go:57]: right now if the name isnt specified aws account name is set and we returning something else from whats submitted is throwing error in terraform apply.

state change detected sent "" got "123456789012"

hence added it as required.

File: api/schema/aws_account_resource.go (hunk 1 above)
Comment (shireesh-illumio) [reply api/schema/aws_account_resource.go:35]: to better match the db field name, and i felt the documentation says it anyways that its access_mode.

File: api/schema/aws_account_resource.go (hunk 2 above)
Comment (rlenglet) [reply api/schema/aws_account_resource.go:57]: Why 3 though? This should be 1 then, correct?

File: api/schema/aws_account_resource.go (hunk 1 above)
Comment (rlenglet) [reply api/schema/aws_account_resource.go:35]: Did you mean to answer to [the other comment](https://github.com/illumio/terraform-provider-illumio-cloudsecure/pull/81#discussion_r1775774952)?

File: api/schema/aws_account_resource.go (hunk 6 above)
Comment (shireesh-illumio) [reply api/schema/aws_account_resource.go:59]: thats added to support existing contract. 
the ui calls depend on the existence of the master accountId to show in some ui endpoints.
else it fails. Hence added it.

File: api/schema/aws_account_resource.go (hunk 3 above)
Comment (shireesh-illumio) [reply api/schema/aws_account_resource.go:68]: same as stated above. master account id isnt returned by the integrations read/update apis. hence made it writeonce.

File: api/schema/aws_account_resource.go (hunk 5 above)
Comment (shireesh-illumio) [reply api/schema/aws_account_resource.go:92]: same issue.

File: api/schema/aws_account_resource.go (hunk 4 above)
Comment (shireesh-illumio) [reply api/schema/aws_account_resource.go:80]: same issue.

File: api/schema/aws_account_resource.go
Comment (shireesh-illumio) [reply api/schema/aws_account_resource.go:1]: ok will remove

File: api/schema/aws_account_resource.go (hunk 1 above)
Comment (shireesh-illumio) [reply api/schema/aws_account_resource.go:35]: to better match the db field name, and i felt the documentation says it anyways that its access_mode.

File: api/schema/aws_account_resource.go (hunk 2 above)
Comment (shireesh-illumio) [reply api/schema/aws_account_resource.go:57]: can be 1. but 3 is what i am usually used to seeing in the name fields. I can make it 1

File: api/schema/aws_account_resource.go (hunk 3 above)
Comment (shireesh-illumio) [reply api/schema/aws_account_resource.go:68]: Immutable attribute is asking it to be sent in response of update and read. 
and role_arn, external_id.
accessMode cause we cannot update it as of now.

File: api/schema/aws_account_resource.go (hunk 6 above)
Comment (shireesh-illumio) [reply api/schema/aws_account_resource.go:59]: this field is needed by some ui endpoint to return properly as it checks this field
to identify existence of org or account.

Comment (rlenglet) [reply api/schema/aws_account_resource.go:59]: We will not support organization onboarding for now.
Please remove this attribute.

File: api/schema/aws_account_resource.go (hunk 3 above)
Comment (rlenglet) [reply api/schema/aws_account_resource.go:80]: Yes, this should be included in responses.
Revert this change.

File: examples/resources/illumio-cloudsecure_aws_account/resource.tf [hunk 7]
```diff
@@ -1,10 +1,11 @@
...
-  role_external_id = "eb287482f5824fab8a6988252d56eb6d"
-
-  # Optional attributes
+  account_id                     = "812713887999"
+  name                           = "Test AWS Account"
+  role_arn                       = "arn:aws:iam::812713887999:role/IllumioAccess"
+  role_external_id               = "eb287482f5824fab8a6988252d56eb6d"
   organization_master_account_id = "965208753613"
-  mode                           = "ReadWrite"
+  organization_id                = "o-1234567890"
```
Comment (rlenglet) [inline examples/resources/illumio-cloudsecure_aws_account/resource.tf:7]: Keep attributes in alphabetic order.

File: examples/resources/illumio-cloudsecure_aws_account/resource.tf [hunk 8]
```diff
@@ -1,10 +1,11 @@
...
-  name             = "Test AWS Account"
-  role_arn         = "arn:aws:iam::812713887999:role/IllumioAccess"
-  role_external_id = "eb287482f5824fab8a6988252d56eb6d"
-
-  # Optional attributes
+  account_id                     = "812713887999"
+  name                           = "Test AWS Account"
+  role_arn                       = "arn:aws:iam::812713887999:role/IllumioAccess"
+  role_external_id               = "eb287482f5824fab8a6988252d56eb6d"
   organization_master_account_id = "965208753613"
```
Comment (rlenglet) [inline examples/resources/illumio-cloudsecure_aws_account/resource.tf:6]: Remove this.
```suggestion
```

**Review by rlenglet — CHANGES_REQUESTED** [review]
My comments have not been addressed.

File: api/schema/aws_account_resource.go (hunk 3 above)
Comment (rlenglet) [reply api/schema/aws_account_resource.go:80]: I asked you to remove the master account ID (https://github.com/illumio/terraform-provider-illumio-cloudsecure/pull/81#discussion_r1775776454), so it doesn't matter whether the backend APIs return that attribute or not.

File: api/schema/aws_account_resource.go (hunk 5 above)
Comment (shireesh-illumio) [reply api/schema/aws_account_resource.go:104]: role_external_id is a secret. writeonly once. we never expect it back from server.

Comment (rlenglet) [reply api/schema/aws_account_resource.go:104]: OK. Fair enough for this one.

Comment (rlenglet) [reply api/schema/aws_account_resource.go:104]: But if we want to treat this as a secret, it must be made `Sensitive: true`. Please do that.

**Comment by rlenglet** [comment]
Separate PR with the same changes and all my comments addressed: https://github.com/illumio/terraform-provider-illumio-cloudsecure/pull/82
Please consider reviewing that instead of doing the changes in this PR.

File: api/illumio/cloud/config/v1/tags.json [hunk 9]
```diff
@@ -1 +1 @@
-{"resource/aws_account":{"account_id":2,"account_type":3,"id":1,"management_account_id":4,"mode":5,"name":6,"organization_account_id":11,"organization_id":7,"organization_master_account_id":12,"role_arn":8,"role_external_id":9,"update_mask":10},"resource/aws_organization":{"account_id":2,"id":1,"master_account_id":9,"mode":3,"name":4,"organization_id":5,"role_arn":6,"role_external_id":7,"update_mask":8},"resource/k8s_cluster_onboarding_credential":{"client_id":2,"client_secret":3,"created_at":4,"description":5,"id":1,"illumio_region":6,"name":7,"update_mask":8}}
\ No newline at end of file
+{"resource/aws_account":{"access_mode":13,"account_id":2,"account_type":3,"id":1,"management_account_id":4,"mode":5,"name":6,"organization_account_id":11,"organization_id":7,"organization_master_account_id":12,"role_arn":8,"role_external_id":9,"update_mask":10},"resource/aws_organization":{"account_id":2,"id":1,"master_account_id":9,"mode":3,"name":4,"organization_id":5,"role_arn":6,"role_external_id":7,"update_mask":8},"resource/k8s_cluster_onboarding_credential":{"client_id":2,"client_secret":3,"created_at":4,"description":5,"id":1,"illumio_region":6,"name":7,"update_mask":8}}
```
Comment (rlenglet) [inline api/illumio/cloud/config/v1/tags.json:1]: Remove the tags for `resource/aws_organization`.
